chainer/link.py

```diff
@@ -328,7 +328,9 @@ def to_gpu(self, device=None):
         if not self._cpu:
             return self
         d = self.__dict__
-        with cuda.get_device_from_id(device.id if device else None):
+        if not device_id:
```

Comment:
And can you make a test for such a case?  I mean a test which calls `to_gpu(0)` with calling `use` with device `1`.